### ISSUE TITLE
Why is my task so affected by the crop size?

### ISSUE THREAD

**Issue by SKBL5694** [open-mmlab/mmaction2#443]
I have trained my net using my own task which is just used to distinguish between someone sleeping or awake.Then I found it works well in my training and testing set(the same input size like 469 * 657 ). And the winner score usually between (0.7 ~ 0.99). But once I change the input size, for example, I crop the frames from the same video at the same time point, but this time the frames are bigger(631 * 664). Though the second frames are bigger than the first, the person I want to distinguish sleeping or awake is still in both frames. This time my net doesn't work well. A certain result(awake) won the game in the classification with an overwhelming score(like 0.99 in softmax). It seems like my net is so affected by input size. By the way, I use the same training and testing pipeline as your configs. Can you give me some suggestions for the situation I encountered?

**Comment by innerlee** [comment]
Testing works best when the data distribution is the same with training data distribution.

To make the network robust to scales, you may randomly resize videos during training.

**Comment by SKBL5694** [comment]
> Testing works best when the data distribution is the same with training data distribution.
> 
> To make the network robust to scales, you may randomly resize videos during training.

From my experience, the official configuration is generally relatively stable, and I found that your configs has done Resize and  MultiScaleCrop. Do you mean I should try to add more data augmentations in  pre-processing data?

**Comment by SuX97** [comment]
Hi, @SKBL5694 :
Convolution operation only has the property of linear translation invariance, but when you change the resolution, the distribution of pixels changes, surely it will cause the gap between the distribution learned by your kernels and the images.

My suggestion for you using larger frames is that, you may stick to (631 * 664), but using a multiple-crop such as TenCrop and crop it to the training resolution i.e. 469 * 657.

**Comment by SKBL5694** [comment]
> Hi, @SKBL5694 :
> Convolution operation only has the property of linear translation invariance, but when you change the resolution, the distribution of pixels changes, surely it will cause the gap between the distribution learned by your kernels and the images.
> 
> My suggestion for you using larger frames is that, you may stick to (631 * 664), but using a multiple-crop such as TenCrop and crop it to the training resolution i.e. 469 * 657.

Thank you for your reply. So what I understand is to train with a large size(maybe using multiple-crop methods) and test with a small size rather than train with small and test with large.

**Comment by SuX97** [comment]
> Thank you for your reply. So what I understand is to train with a large size(maybe using multiple-crop methods) and test with a small size rather than train with small and test with large.

Right, you can either: 
1. train with large images and test with the same resolution
2. keep using the model trained with small images, but first crop (631 * 664) -> (469 * 657), and resize (469 * 657) -> (-1, 256), then crop (-1, 256)->(256, 256) -> inference.

In one word, try to train and test with the same resolution.

**Comment by SKBL5694** [comment]
> > Thank you for your reply. So what I understand is to train with a large size(maybe using multiple-crop methods) and test with a small size rather than train with small and test with large.
> 
> Right, you can either:
> 
> 1. train with large images and test with the same resolution
> 2. keep using the model trained with small images, but first crop (631 * 664) -> (469 * 657), and resize (469 * 657) -> (-1, 256), then crop (-1, 256)->(256, 256) -> inference.
> 
> In one word, try to train and test with the same resolution.

I get it. Thx :)
